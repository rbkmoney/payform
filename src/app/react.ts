import 'react';
import 'react-dom';
import 'react-transition-group';
import 'react-redux';
import 'redux-form';
import 'redux-thunk';
import 'redux-saga';
import 'classnames';
