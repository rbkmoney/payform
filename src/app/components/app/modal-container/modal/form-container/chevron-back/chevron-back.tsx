import * as React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';
import * as formStyles from '../form-container.scss';
import { setFormFlowAction, SetFormsFlowAction } from 'checkout/actions';
import { FormFlowItem, State } from 'checkout/state';
import { Icon, IconType } from 'checkout/components';
import { back } from '../form-flow-manager';

export interface ChevronBackProps {
    formsFlow: FormFlowItem[];
    setFormFlow: (formFlow: FormFlowItem[]) => SetFormsFlowAction;
}

const mapStateToProps = (state: State) => ({
    formsFlow: state.formsFlow
});

const mapDispatchToProps = (dispatch: Dispatch<any>) => ({
    setFormFlow: bindActionCreators(setFormFlowAction, dispatch)
});

class ChevronBackDef extends React.Component<ChevronBackProps> {

    constructor(props: ChevronBackProps) {
        super(props);
        this.back = this.back.bind(this);
    }

    back() {
        this.props.setFormFlow(back(this.props.formsFlow));
    }

    render() {
        return (
            <div className={formStyles.back_btn} onClick={this.back}>
                <Icon icon={IconType['chevron-left']}/>
            </div>
        );
    }
}

export const ChevronBack = connect(mapStateToProps, mapDispatchToProps)(ChevronBackDef);
