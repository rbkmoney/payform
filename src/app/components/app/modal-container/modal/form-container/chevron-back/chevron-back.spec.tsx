import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { ChevronBack } from './chevron-back';

it('renders correctly', () => {
    const tree = renderer
        .create(<ChevronBack className={'test-class'} back={() => null} id={'test-id'}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
