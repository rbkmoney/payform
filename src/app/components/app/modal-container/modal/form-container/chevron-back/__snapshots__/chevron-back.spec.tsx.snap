// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="test-class"
  id="test-id"
  onClick={[Function]}
>
  <svg
    className={undefined}
  >
    <use
      xlinkHref="assets/icons/chevron-left.svg#icon"
    />
  </svg>
</div>
`;
